src/common/transformations/src/transformations/common_optimizations/sequence_fusion.cpp

```diff
@@ -0,0 +1,324 @@
+               cell_1->get_activations_beta() == cell_2->get_activations_beta() &&
+               cell_1->get_clip() == cell_2->get_clip() && check_WRB(cell_1, cell_2);
+    return is_equal;
+}
+
+shared_ptr<RNNCellBase> find_cell_chain(ov::pass::NodeRegistry& cp_from,
+                                        ov::pass::NodeRegistry& cp_to,
+                                        const shared_ptr<RNNCellBase>& current_cell,
+                                        ov::OutputVector& x_to_concat,
+                                        ov::OutputVector& attention_to_concat,
+                                        map<int, ov::Output<ov::Node>>& h_outputs_to_redirect,
+                                        int& cells_cnt,
+                                        const shared_ptr<ov::Node>& axis_1) {
+    cells_cnt = 1;
+
+    shared_ptr<RNNCellBase> current = current_cell;
+    while (true) {
+        cp_from.add(current);
+        // check the source node of HiddenState input
+        auto prev = current->input_value(1).get_node_shared_ptr();
+        if (auto prev_cell = dynamic_pointer_cast<RNNCellBase>(prev)) {
+            if (!is_equal_cells(prev_cell, current)) {
+                break;
+            }
+
+            auto in_X = current->input(0);
+            x_to_concat.push_back(cp_to.make<Unsqueeze>(in_X.get_source_output(), axis_1));
+            h_outputs_to_redirect[cells_cnt] = current->output(0);
+
+            // check intermediate C outputs in case of LSTMCell
+            if (dynamic_pointer_cast<LSTMCell>(current) && cells_cnt != 1) {
+                const auto& target_inputs = current->get_output_target_inputs(1);
+                bool valid = target_inputs.empty() ||
+                             (target_inputs.size() == 1 && dynamic_cast<LSTMCell*>(target_inputs.begin()->get_node()) &&
+                              target_inputs.begin()->get_index() == 2);
+                if (!valid) {
+                    // if intermediate C output is connected to other node, except LSTMCell,
+                    // we can't replace cells with sequence. Sequence doesn't provide access to these outputs.
+                    return {};
+                }
+            }
+
+            if (auto augru = dynamic_pointer_cast<ov::op::internal::AUGRUCell>(current)) {
+                attention_to_concat.push_back(cp_to.make<Unsqueeze>(augru->input_value(5), axis_1));
+            }
+
+            current = prev_cell;
+            cells_cnt++;
+        } else {
+            auto in_X = current->input(0);
+            x_to_concat.push_back(cp_to.make<Unsqueeze>(in_X.get_source_output(), axis_1));
+            if (auto augru = dynamic_pointer_cast<ov::op::internal::AUGRUCell>(current)) {
+                attention_to_concat.push_back(cp_to.make<Unsqueeze>(augru->input_value(5), axis_1));
+            }
+            h_outputs_to_redirect[cells_cnt] = current->output(0);
+            break;
+        }
```

Comment:
Looks like this piece of code in the `else` block could be moved out of the `if/else` as it is repeated twice.
https://github.com/openvinotoolkit/openvino/blob/20508c9aeb741bb767f25b1566273fa7dac3bfbd/src/common/transformations/src/transformations/common_optimizations/sequence_fusion.cpp#L96-L98 https://github.com/openvinotoolkit/openvino/blob/20508c9aeb741bb767f25b1566273fa7dac3bfbd/src/common/transformations/src/transformations/common_optimizations/sequence_fusion.cpp#L113-L115

Reply:
done